llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -3245,6 +3246,30 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
         MaybeSimplifyHint(OBU.Inputs[0]);
         MaybeSimplifyHint(OBU.Inputs[1]);
       }
+
+      // Try to fold alignment assumption into a load's !align metadata, if the
+      // assumption is valid in the load's context and remove redundant ones.
```

Comment:
I'm a bit confused -- the code doesn't seem to do anything with load metadata?

Reply:
Yeah the comment was left over from a different PR built on top of this one. Removed.